modules/nw-ingress-sharding.adoc

```diff
@@ -1,17 +1,105 @@
 // Module included in the following assemblies:
 //
-// * configuring_ingress_cluster_traffic/configuring-ingress-cluster-traffic-ingress-controller.adoc
 // * ingress-operator.adoc
+// * networking/ingress-sharding.adoc
 
+:_content-type: CONCEPT
 [id="nw-ingress-sharding_{context}"]
 = Ingress Controller sharding
 
-As the primary mechanism for traffic to enter the cluster, the demands on the Ingress Controller, or router, can be significant. As a cluster administrator, you can shard the routes to:
+Ingress sharding, also known as router sharding, lets you distribute a set of routes across multiple routers by adding labels to routes, namespaces, or both. The Ingress Controller uses a corresponding set of selectors to admit only the routes with a specified label. Each Ingress shard comprises the routes that are filitered using a given selection expression.
+
+As the primary mechanism for traffic to enter the cluster, the demands on the Ingress Controller can be significant. As a cluster administrator, you can shard the routes to:
 
 * Balance Ingress Controllers, or routers, with several routes to speed up responses to changes.
 * Allocate certain routes to have different reliability guarantees than other routes.
 * Allow certain Ingress Controllers to have different policies defined.
 * Allow only specific routes to use additional features.
 * Expose different routes on different addresses so that internal and external users can see different routes, for example.
+* Transfer traffic from one version of an application to another during a blue green deployment.
+
+When Ingress Controllers are sharded, a given route is admitted to zero or more Ingress Controllers in the group. A route's status describes whether an Ingress Controller has admitted it or not. An Ingress Controller will only admit a route if it is unique to its shard.
+
+An Ingress Controller can use three sharding methods:
+
+. Adding only a namespace selector to the Ingress Controller, so that all routes in a namespace with labels that match the namespace selector are in the Ingress shard.
+
+. Adding only a route selector to the Ingress Controller, so that all routes with labels that match the route selector are in the Ingress shard.
+
+. Adding both a namespace selector and route selector to the Ingress Controller, so that routes with labels that match the route selector in a namespace with labels that match the namespace selector are in the Ingress shard.
+
+This lets you distribute subsets of routes over multiple Ingress Controllers. These subsets can be non-overlapping, also called _traditional_ sharding, or overlapping, otherwise known as _overlapped_ sharding.
+
+== Traditional sharding example
```

Comment:
These examples could be broken out into reference modules if you liked.

Reply:
This is a good idea, thanks! I think they could be tweaked a little more, so I'll open a followup PR for modularization